nl/seg6_linux.go

```diff
@@ -0,0 +1,121 @@
+package nl
+
+import (
+	"errors"
+	"fmt"
+	"net"
+)
+
+type Ipv6SrHdr struct {
+	nexthdr       uint8
+	hdrlen        uint8
+	typ           uint8
```

Comment:
Minor: To avoid using the reserved `type` string, can you please rename this field to `routingType` ?
I know it is an internal field, but I am not sure `typ` would directly convey what it represents to somebody reading the code.

Reply:
I followed what's used as temp var in MPLS code.
But yes, `routingType` should be easier to understand as a name of `IPv6SrHdr` member.
I changed ones in `nl/seg6_linux.go` to `routingType`.
I kept ones in `route_linux.go` as is `typ` since they are temp var used very close to where defined by `:=`.